/*
 * This file was automatically generated by SmartUnit
 */

package smartunit.com.macro.mall.common.exception;

import org.junit.Test;
import static org.junit.Assert.*;
import java.lang.reflect.Method;
import static org.smartunit.shaded.org.mockito.Mockito.*;
import org.smartunit.shaded.org.mockito.ArgumentMatchers;
import static org.smartunit.runtime.SmartAssertions.*;
import com.macro.mall.common.api.IErrorCode;
import com.macro.mall.common.exception.Asserts;
import org.junit.runner.RunWith;
import org.smartunit.runtime.ExecutorServiceAnswer;
import org.smartunit.runtime.SmartRunner;
import org.smartunit.runtime.SmartRunnerParameters;
import org.smartunit.runtime.TransactionTemplateAnswer;
import org.smartunit.runtime.ViolatedAssumptionAnswer;

@RunWith(SmartRunner.class) @SmartRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class Asserts_SSTest extends Asserts_SSTest_scaffolding {
// allCoveredLines:[9, 11, 15]

  @Test(timeout = 4000)
  public void test_fail_0()  throws Throwable  {
      //caseID:a84cf309368a28f2352eb2eea3e76dc8
      //CoveredLines: [15]
      //Input_0_IErrorCode: {}
      
      //mock iErrorCode0
      IErrorCode iErrorCode0 = mock(IErrorCode.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      
      //Call method: fail
      // Undeclared exception!
      try { 
        Asserts.fail(iErrorCode0);
      } catch(Throwable e) {
         verifyException("com.macro.mall.common.exception.Asserts", e);
         assertEquals("com.macro.mall.common.exception.ApiException", e.getClass().getName());
      }
  }

  @Test(timeout = 4000)
  public void test_fail_1()  throws Throwable  {
      //caseID:bf48724d2765f6331116fd29f46d8653
      //CoveredLines: [9, 11]
      //Input_0_String: 
      
      Asserts asserts0 = new Asserts();
      
      //Call method: fail
      // Undeclared exception!
      try { 
        Asserts.fail("");
      } catch(Throwable e) {
         verifyException("com.macro.mall.common.exception.Asserts", e);
         assertEquals("com.macro.mall.common.exception.ApiException", e.getClass().getName());
      }
  }
}
